opentreemap/treemap/templatetags/auth_extras.py

```diff
@@ -215,3 +224,28 @@ def render(self, context):
         # If there was a user match, the function would have
         # previously returned the protected content
         return ''
+
+
+@register.simple_tag(takes_context=True)
+def login_forward(context):
+    """
+    If the current page is an instance page and the user is not logged in,
+    return the `?next=` query param with a value that is the sanitized
+    version of the current page url.
+
+    `login_forward` should not be called if the user is already logged in.
+
+    Return an empty string if the current page is not an instance page.
+    """
+    request = template.Variable('request').resolve(context)
+
+    if getattr(request, 'user', None) and request.user.is_authenticated():
+        raise ValidationError(
+            _('Can\'t forward login if already logged in'))
+    if not getattr(request, 'instance', None):
+        return ''  # No point in forwarding back to a static page
```

Comment:
Could you elaborate on why there is no point in forwarding back to a non-instance page? If I am on `/users/{username}` and click "Login" it would be nice and consistent to return to the profile page after logging in. Does including non-instance pages cause more trouble than it is worth?